### PR TITLE
 Install podman-systemd.unit man page, make quadlet discoverable 

### PR DESCRIPTION
**Pull request by rhatdan**
Currently we are shipping no data about quadlet, since the
podman-systemd.unit file is not shipped. Also want to add the
quadlet name to the description of the man page so that

man -k quadlet

will help users find the man page.

Also add a link file such that if the user types in
man quadlet
man will show the podman-systemd.unit file.

Also eliminate the subpackage podman-quadlet

Fixes: https://github.com/containers/podman/issues/17349

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Add podman-systemd.unit man page
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@lsm5 PTAL
@vrothberg @mheon @alexlarsson @ygalblum @baude PTAL

This needs to be added to podman-4.4.1

**Comment by rhatdan** [comment]
@lsm5 Could you verify that I did the podman.spec.rpkg correctly

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
I want to remove the podman-quadlet subpackage. it should be installed by default and is not big enough to require a separate package.

@lsm5 I probably screwed up the spec file.

File: podman.spec.rpkg
Comment (lsm5) [inline podman.spec.rpkg:235]: these are easier:
```
%{_systemdgeneratordir}/%{name}-system-generator
%{_systemdusergeneratordir}/%{name}-user-generator
```

I don't know if rhel 8 and 9 have these defined already, let me take a quick look and get back.

Comment (lsm5) [reply podman.spec.rpkg:235]: Yes, these 2 macros should work on rhel8 and 9.

File: podman.spec.rpkg
Comment (lsm5) [inline podman.spec.rpkg:87]: We should also have a `Provides: %{name}-quadlet` here if we are removing the subpackage. An obsoletes would also be nice, but I gotta figure out the last build that shipped a separate podman-quadlet.

**Comment by lsm5** [comment]
> I want to remove the podman-quadlet subpackage. it should be installed by default and is not big enough to require a separate package.
> 
> @lsm5 I probably screwed up the spec file.

I don't see any additional systemd unit installed, there's only the installation of the unitfile manpage, among other things.

File: podman.spec.rpkg
Comment (lsm5) [reply podman.spec.rpkg:87]: Let me disable podman autobuilds and get you the last NVR to add here.

Comment (lsm5) [reply podman.spec.rpkg:87]: Add these as well.
```
Obsoletes: %{name}-quadlet <= 101:0.0.git.17877.f247b4d4-1
Provides: %{name}-quadlet = %{epoch}:%{version}-%{release}
```

I will resume podman autobuilds after this merges.

**Comment by lsm5** [comment]
@rhatdan so looks like I should wait for v4.4.1 to move quadlet files into the main package on Fedora?

**Comment by rhatdan** [comment]
@lsm5 PTANOTHERLOOK

**Review by vrothberg — APPROVED** [review]
LGTM

File: podman.spec.rpkg
Comment (lsm5) [reply podman.spec.rpkg:87]: @rhatdan could you add these please? That should keep upgrades smooth.

**Review by lsm5 — CHANGES_REQUESTED** [review]
Good to go after https://github.com/containers/podman/pull/17351/files#r1095784822 is addressed.

File: podman.spec.rpkg
Comment (rhatdan) [reply podman.spec.rpkg:87]: It has been addressed?  See Below?

Comment (lsm5) [reply podman.spec.rpkg:87]: ah, my bad. LGTM

**Review by lsm5 — APPROVED** [review]
/lgtm
/hold

**Comment by rhatdan** [comment]
Since this was addressed, I am going to merge.

**Comment by lsm5** [comment]
rebuilds enabled.

**Pull request merged**
